### PR TITLE
Recruitment web app - Account #11

### PR DESCRIPTION
**Pull request by longlh-dev**
issue 11: https://github.com/HopeeSolutionsRuby/demo-app/issues/11
description: 
- setup account file using devise gem
- upload avatar feature using carrierwave gem
- implement redirect after login for each roles

### CODE REVIEW

File: app/controllers/admin/dashboard_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Admin
```
Comment (KhoaNDAHopee) [inline app/controllers/admin/dashboard_controller.rb]: Should use plural

File: app/controllers/employee/dashboard_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Employee
```
Comment (KhoaNDAHopee) [inline app/controllers/employee/dashboard_controller.rb]: Should use plural

File: app/controllers/employer/dashboard_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+module Employer
```
Comment (KhoaNDAHopee) [inline app/controllers/employer/dashboard_controller.rb]: Should use plural

File: app/models/account.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+         :recoverable, :rememberable, :validatable
+  validates :phone, phone: true, uniqueness: true, unless: -> { phone.blank? }
+
+  enum role: { employee: 0, employer: 1, admin: 2 }
+
+  mount_uploader :avatar, AvatarUploader
+
+  private
+
+  def downcase_email
```
Comment (KhoaNDAHopee) [inline app/models/account.rb]: Research about devise configs ``config.case_insensitive_keys = [:email]``

File: app/uploaders/avatar_uploader.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class AvatarUploader < CarrierWave::Uploader::Base
+  # Include RMagick or MiniMagick support:
+  # include CarrierWave::RMagick
+  include CarrierWave::MiniMagick
+
+  # Choose what kind of storage to use for this uploader:
+  storage :file
```
Comment (KhoaNDAHopee) [inline app/uploaders/avatar_uploader.rb]: Set global config ``config/initializers/carrierwave.rb``

File: config/environments/development.rb [hunk 6]
```diff
@@ -8,6 +8,7 @@
   # In the development environment your application's code is reloaded any time
   # it changes. This slows down response time but is perfect for development
   # since you don't have to restart the web server when you make code changes.
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
```
Comment (KhoaNDAHopee) [inline config/environments/development.rb:11]: Set configs for https://github.com/ryanb/letter_opener, https://github.com/fgrehm/letter_opener_web
(fake mail server for development)

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]
- Check conflict ruby version in Gemfile, .ruby-version
- Implement ``forgot password``
- Show error when input invalid email/password

**Comment by DucNguyen0812** [comment]
LGTM!

**Review by DucNguyen0812 — APPROVED** [review]

**Review by Priyarekah — APPROVED** [review]
I tested it and it is fine with me.

**Review by KhoaNDAHopee — APPROVED** [review]
LGTM

**Pull request merged**